Documentation for FLUIDOS Node Implementation (#7)
fluidos-project/node: docs/implementation/components.md

Reviewer: Let's see how each component has been implemented*

The primary objectives of the Discovery Manager are as follows:

- **Populating Peering Candidates Table**: The Discovery Manager's primary responsibility is to populate the Peering Candidates table. It achieves this by identifying suitable resources known as "Flavours" based on the initial messages exchanged as part of the REAR protocol.

- **Offering Appropriate Flavours**: If no suitable peering candidates are initially available, the REAR Manager constructs a flavor selector and forwards it to the Discovery Manager. The Discovery Manager, in turn, initiates a LIST_FLAVOURS message, broadcasting it to all known endpoints, including local Nodes, Supernodes, and Catalogs.

The Discovery Manager plays a pivotal role in facilitating resource discovery and allocation within the FLUIDOS ecosystem. It contributes to the seamless orchestration of resources across FLUIDOS Nodes, ensuring efficient utilization and allocation of computing resources.

### Peering Candidates

> The **Peering Candidates** component manages a dynamic list of nodes that are potentially suitable for establishing peering connections. This list is continuously updated based on the available resources in the nodes and the requests for flavours from the Discovery Manager.

Author: ```suggestion
The **Peering Candidates** component manages a dynamic list of nodes that are potentially suitable for establishing peering connections. This list is continuously updated by the Discovery Manager.
```

> Under the hood, Peering Candidates seamlessly integrates with Kubernetes' `etcd`, ensuring efficient storage and retrieval of data.

Author: ```suggestion
Under the hood, Peering Candidates are stored through an appropriate Custom Resource. 
```

### REAR Manager

> The **REAR Manager** plays a pivotal role in orchestrating the service provisioning process. It receives service requests, translates them into resource or service requests, and looks up suitable resources. 

Author: ```suggestion
The **REAR Manager** plays a pivotal role in orchestrating the service provisioning process. It receives a solving request, translates them into resource or service requests, and looks up external suitable resources. 
```

> If none are found, it initiates the Discovery, Reservation, and Peering phases. It selects potential peering candidates and can request discovery. The suitable local flavors are identified and shared. If a suitable candidate is found, it triggers the Reservation phase. Resources are allocated, contracts are created, and peering is established.

Author: ```suggestion
If none are found, it initiates the Discovery, Reservation, and Peering phases. It can select potential peering candidates or, optionally, request discovery. If a suitable candidate is found, it triggers the Reservation phase. If this process is successfully fulfilled, resources are allocated, contracts are stored, and peering is established.
```
